tinygrad/codegen/uops.py

```diff
@@ -151,17 +151,19 @@ def print(self):
 
   def add(self, uop:UOps, dtype:Optional[DType]=None, vin:Tuple[UOp, ...]=tuple(), arg:Any=None, cachable=True, insert_before=None,
           simplify=True) -> UOp:
-    ret = UOp(uop, dtype, vin, arg) if uop is not UOps.CONST else UOp.const(dtype, arg)
-    if simplify and (rewritten:=constant_folder.rewrite(ret)) is not None:
+    return self.add_op(UOp(uop, dtype, vin, arg) if uop is not UOps.CONST else UOp.const(dtype, arg), cachable, insert_before, simplify)
+
+  def add_op(self, op:UOp, cachable=True, insert_before=None, simplify=True) -> UOp:
+    if simplify and (rewritten:=constant_folder.rewrite(op)) is not None:
       if rewritten in self.uops: return rewritten  # ignore cachable
-      ret = rewritten
```

Comment:
This rename shouldn't be mixed In with functional changes, it makes review hard.